docs/ibc/light-clients/wasm/concepts.md

```diff
@@ -0,0 +1,71 @@
+- `LatestHeight` is the latest height of the counterparty state machine (i.e. the height of the blockchain), whose consensus state the light client tracks.
+
+```go
+type ClientState struct {
+  // bytes encoding the client state of the underlying 
+  // light client implemented as a Wasm contract
+  Data         []byte
+  // sha256 hash of Wasm contract bytecode
+  CodeHash     []byte
+  // latest height of the counterparty ledger
+  LatestHeight types.Height
+}
+```
+
+See section [`ClientState` of the light client module developer guide](../overview.md#clientstate) for more information about the `ClientState` interface.
+
+### `ConsensusState`
+
+The `08-wasm`'s `ConsensusState` data structure maintains one field:
+
+- `Data` contains the bytes of the Protobuf-encoded consensus state of the underlying light client implemented as a Wasm contract. For example, if the light client Wasm contract implements the Tendermint light client algorithm, then `Data` will contain the bytes for a [Tendermint consensus state](https://github.com/cosmos/ibc-go/blob/v7.2.0/modules/light-clients/07-tendermint/tendermint.pb.go#L101-L109).
+
+```go
+type ConsensusState struct {
+  // bytes encoding the consensus state of the underlying light client
+  // implemented as a Wasm contract.
+  Data []byte
+}
+```
+
+See section [`ConsensusState` of the light client module developer guide](../overview.md#consensusstate) for more information about the `ConsensusState` interface.
+
+### `ClientMessage`
+
+`ClientMessage` is used for performing updates to a `ClientState` stored on chain. The `08-wasm`'s `ClientMessage` data structure maintains one field:
+
+- `Data` contains the bytes of the Protobuf-encoded header(s) or misbehaviour for the underlying light client implemented as a Wasm contract. For example, if the light client Wasm contract implements the Tendermint light client algorithm, then `Data` will contain the bytes of either [header](https://github.com/cosmos/ibc-go/blob/v7.2.0/modules/light-clients/07-tendermint/tendermint.pb.go#L186-L203) or [misbehaviour](https://github.com/cosmos/ibc-go/blob/v7.2.0/modules/light-clients/07-tendermint/tendermint.pb.go#L144-L151) for a Tendermint light client.
+
+```go
+type ClientMessage struct {
+  // bytes encoding the header(s) or misbehaviour for the underlying light client
+  // implemented as a Wasm contract.
+  Data []byte
+}
+```
+
+See section [`ClientMessage` of the light client module developer guide](../overview.md#clientmessage) for more information about the `ClientMessage` interface.
+
+## Wasm light client
+
+The actual light client can be implemented in any language that compiles to Wasm and implements the interfaces of a [CosmWasm](https://docs.cosmwasm.com/docs/) contract. Even though in theory other languages could be used, in practic (at least for the time being) the most suitable language to use would be Rust, since there is already good support for it for developing CosmWasm smart contracts.
```

Comment:
```suggestion
The actual light client can be implemented in any language that compiles to Wasm and implements the interfaces of a [CosmWasm](https://docs.cosmwasm.com/docs/) contract. Even though in theory other languages could be used, in practice (at least for the time being) the most suitable language to use would be Rust, since there is already good support for it for developing CosmWasm smart contracts.
```